src/Http/Deserializer/Calendar/CalendarJSONParser.php

```diff
@@ -36,35 +28,46 @@ public function getEndDate($data)
         return new \DateTime($data['endDate']);
     }
 
+    public function getStatus(array $data): ?Status
+    {
+        if (!isset($data['status'])) {
+            return null;
+        }
+
+        return Status::deserialize($data['status']);
+    }
+
     /**
-     * @param array $data
-     *
-     * @return TimeSpan[]
+     * @return Timestamp[]
      */
-    public function getTimeSpans($data)
+    public function getTimestamps(array $data): array
     {
-        $timeSpans = [];
+        $timestamps = [];
 
         if (!empty($data['timeSpans'])) {
             foreach ($data['timeSpans'] as $index => $timeSpan) {
                 if (!empty($timeSpan['start']) && !empty($timeSpan['end'])) {
                     $startDate = new \DateTime($timeSpan['start']);
                     $endDate = new \DateTime($timeSpan['end']);
-                    $timeSpans[] = new TimeSpan($startDate, $endDate);
+                    $timestamp = new Timestamp($startDate, $endDate);
+
+                    if (isset($timeSpan['status'])) {
+                        $timestamp = $timestamp->withStatus(Status::deserialize($timeSpan['status']));
+                    }
+
+                    $timestamps[] = $timestamp;
```

Comment:
Same nit: Some early returns (or in the case of the `foreach` a `continue`) could reduce the amount of indentations here.

Reply:
That makes the code indeed much cleaner.